src/Features/Core/Portable/Testing/AbstractTestMethodFinder.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Features.Testing;
+
+internal abstract class AbstractTestMethodFinder<TMethodDeclaration>(IEnumerable<ITestFrameworkMetadata> testFrameworks) : ITestMethodFinder where TMethodDeclaration : SyntaxNode
+{
+    /// <summary>
+    /// Output the method symbol as a fully qualified method name, e.g. Namespace.Class.Method to match what test discovery gives us.
+    /// Generics are not applicable here - none of our supported test frameworks allow generic test classes / methods.
+    /// </summary>
+    private static readonly SymbolDisplayFormat s_methodSymbolNoParametersDisplayFormat = new(
+        typeQualificationStyle: SymbolDisplayTypeQualificationStyle.NameAndContainingTypesAndNamespaces,
+        memberOptions: SymbolDisplayMemberOptions.IncludeContainingType);
+
+    /// <summary>
+    /// Sometimes the test runner outputs fully qualified method names with parameters.  So we also check for that scenario.
```

Comment:
does this work for how the test runner might output the FQN for a type like System.Int32?

Reply:
Even worse.  This varies based on the test runner.  If you have an integer parameter the test runner sometimes outputs something like this `NUnitTestProject.Tests.Test2(2)` (where 2 is the theory data) as the FQN.  

I'm going to modify this to instead always look at the non-parameterized form and strip off anything after the last `(`.